### ISSUE TITLE
Need pad_token_id?

### ISSUE THREAD

**Issue by ShiJiawenwen** [imoneoi/openchat#131]
Thanks for your great work! I know openchat-3.5 use the padding-free training strategy. May I ask if it is necessary to add pad_token during the inference stage? 

The example code is as follows:

tokenizer = AutoTokenizer.from_pretrained("openchat/openchat_3.5")
tokenizer.pad_token="<|pad_0|>"

When I use this model for generation get the warning message: 
"The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:32000 for open-end generation."

**Comment by imoneoi** [comment]
During inference, pad should be set to id `0` (the original PAD token of the base model). However, if properly masked, padding won't have much impact on performance.

**Comment by ShiJiawenwen** [comment]
I want to know how to correct masked, could you provide the example code？I greatly appreciate your help!

**Comment by imoneoi** [comment]
Mask all attention and loss in pad tokens. We use flash attention varlen API and set loss weight in pad tokens to 0.

https://github.com/imoneoi/openchat/blob/master/ochat/models/unpadded_mistral.py#L185
https://github.com/imoneoi/openchat/blob/master/ochat/training_deepspeed/openchat_dataset.py#L80

**Comment by ShiJiawenwen** [comment]
hello, I get the new question. Is the embedding of inputs can get like other models use huggingface? like:

def get_embeddings(model, input_ids):
        return model.model.embed_tokens(input_ids)

embeds = get_embeddings(model, input_ids.unsqueeze(0)).detach()
